python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -416,7 +513,7 @@ def launch_scheduled_runs_for_schedule(
                             # or the schedule is turned off
                             failure_count=tick_context.failure_count,
                         )
-                        raise  # Raise the wrapped DagsterSchedulerError exception
+                        yield error_data
```

Comment:
i assume if the tests all pass this change is fine but its a little spooky to me - was this just cargo cult or did you have intention behind it?

Reply:
I had intention - certain tests were failing because errors were not propogating upward properly in the multithreaded case if they were just raised instead of the error data yielded - for example `test_bad_env_fn_with_retries`. The one thing that could be spooky is if more code got added later - after this except block, that code could potentially run. So maybe we want to return post-yield.